### PR TITLE
Fixes #35013 - config rake uses Setting DSL

### PR DESCRIPTION
**Pull request by ezr-ondrej**
config rake was still using obsolete setting access.

### CODE REVIEW

**Review by ehelms — DISMISSED** [review]
Can confirm it fixes the pipeline problem that identified this:

```
# /usr/sbin/foreman-rake -- config -k 'pulp_client_cert' -v '/etc/pki/katello/certs/pulp-client.crt'
pulp_client_cert: /etc/pki/katello/certs/pulp-client.crt
```

Thanks @ezr-ondrej !

File: lib/tasks/config.rake [hunk 1]
```diff
@@ -91,30 +91,34 @@ task :config => :environment do
...
+        setting = Foreman.settings.find(key)
         if value == :unset
-          setting.value = nil
+          val = nil
         elsif complex_type?(setting.settings_type)
-          setting.value = typecast_value(setting.settings_type, value)
+          val = typecast_value(setting.settings_type, value)
         else
-          parse_and_set_string(setting, value)
+          val = value
```
Comment (ofedoren) [inline lib/tasks/config.rake:108]: Is it okay that we don't use `val` anywhere?.. Or am I missing something.

Comment (ezr-ondrej) [reply lib/tasks/config.rake:108]: It is very not ok :thinking:
Good catch!

Comment (ezr-ondrej) [reply lib/tasks/config.rake:108]: Fixed

**Review by ofedoren — APPROVED** [review]
Thanks, @ezr-ondrej, LGTM. Not sure if Katello test failure is related though...

**Review by evgeni — APPROVED** [review]

**Pull request merged**